cs-amazon/src/main/java/io/cloudslang/content/amazon/actions/cloudformation/ListStacksAction.java

```diff
@@ -0,0 +1,106 @@
+     *                          Default: ""
+     * @param proxyPort         Optional - proxy server port. You must either specify values for both proxyHost and
+     *                          proxyPort inputs or leave them both empty.
+     *                          Default: ""
+     * @param proxyUsername     Optional - proxy server user name.
+     *                          Default: ""
+     * @param proxyPassword     Optional - proxy server password associated with the proxyUsername input value.
+     *                          Default: ""
+     * @param region            AWS region name
+     *                          Example: "eu-central-1"
+     * @return                  A map with strings as keys and strings as values that contains: outcome of the action, returnCode of the
+     *                          operation, or failure message and the exception if there is one
+     */
+    @Action(name = "List AWS Cloud Formation stacks",
+            outputs = {
+                    @Output(Outputs.RETURN_CODE),
+                    @Output(Outputs.RETURN_RESULT),
+                    @Output(Outputs.EXCEPTION)
+            },
+            responses = {
+                    @Response(text = Outputs.SUCCESS, field = Outputs.RETURN_CODE, value = Outputs.SUCCESS_RETURN_CODE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.RESOLVED),
+                    @Response(text = Outputs.FAILURE, field = Outputs.RETURN_CODE, value = Outputs.FAILURE_RETURN_CODE,
+                            matchType = MatchType.COMPARE_EQUAL, responseType = ResponseType.ERROR)
+            }
+    )
+    public Map<String, String> execute(
+            @Param(value = IDENTITY,   required = true)  String identity,
+            @Param(value = CREDENTIAL, required = true, encrypted = true)  String credential,
+            @Param(value = REGION,     required = true)  String region,
+            @Param(value = PROXY_HOST)                   String proxyHost,
+            @Param(value = PROXY_PORT)                   String proxyPort,
+            @Param(value = PROXY_USERNAME)               String proxyUsername,
+            @Param(value = PROXY_PASSWORD)               String proxyPassword,
+            @Param(value = CONNECT_TIMEOUT)              String connectTimeoutMs,
+            @Param(value = EXECUTION_TIMEOUT)            String execTimeoutMs) {
+
+        proxyPort = defaultIfEmpty(proxyPort, DefaultValues.PROXY_PORT);
+        connectTimeoutMs = defaultIfEmpty(connectTimeoutMs, DefaultValues.CONNECT_TIMEOUT);
+        execTimeoutMs = defaultIfBlank(execTimeoutMs, DefaultValues.EXEC_TIMEOUT);
+
+        try {
+            AmazonCloudFormation stackBuilder = CloudFormationClientFactory.getCloudFormationClient(identity, credential, proxyHost, proxyPort, proxyUsername, proxyPassword, connectTimeoutMs, execTimeoutMs, region);
+
+            StringBuilder listOfStacksResult = new StringBuilder(EMPTY);
```

Comment:
All the following logic shouldn't be here... A specialized service with some kind of wrapper it would be better, right?